apps/sms/js/thread_ui.js

```diff
@@ -737,15 +737,18 @@ var ThreadUI = {
       // Otherwise find them from draft recipient numbers
       draft.recipients.forEach(function(number) {
         Contacts.findByAddress(number).then(function(contacts) {
+          var recipient;
           if (contacts.length) {
-            this.recipients.add(
-              Utils.basicContact(number, contacts[0])
-            );
+            recipient = Utils.basicContact(number, contacts[0]);
+            recipient.source = 'contacts';
           } else {
-            this.recipients.add({
-              number: number
-            });
+            recipient = {
+              number: number,
+              source: 'manual'
+            };
           }
+
+          this.recipients.add(recipient);
         }.bind(this));
       }, this);
```

Comment:
Yeah, I think it's a good idea, maybe in follow-up! But what if we rename it to ```Recipient.fromNumber()``` (that requests ```Contacts``` and validate number if needed) and we can use it not only in the ```handleDraft```?